projects/gson/FuzzParse.java

```diff
@@ -0,0 +1,27 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+import com.code_intelligence.jazzer.api.FuzzedDataProvider;
+
+import java.io.*;
+import com.google.gson.*;
+
+public class FuzzParse {
+  public static void fuzzerTestOneInput(FuzzedDataProvider data) {
+    try {
+      JsonParser.parseString(data.consumeRemainingAsString());
+    } catch (JsonSyntaxException expected) { }
```

Comment:
Maybe it would make sense to test JSON parsing in lenient and non-lenient mode. Unfortunately Gson does not respect the lenientness setting in some cases for historical reasons, so the following might be an alternative:
```java
TypeAdapter<JsonElement> adapter = new Gson().getAdapter(JsonElement.class);
JsonReader reader = new JsonReader(new StringReader(#INPUT#));
reader.setLenient(#LENIENT#);
while (reader.peek() != JsonToken.END_DOCUMENT) {
    adapter.read(reader);
}
```
(`#INPUT#` and `#LENIENT#` are placeholders here)
This also makes sure that the end of the JSON document is reached, and that in lenient mode multiple top level JSON elements are parsed.

----

Note that I am not a member of Gson.

Reply:
Added a new fuzzer following this paradigm.  Thank you!